adapters/visx/visx.go

```diff
@@ -117,9 +131,19 @@ func (a *VisxAdapter) MakeBids(internalRequest *openrtb2.BidRequest, externalReq
 
 }
 
-func getMediaTypeForImp(impID string, imps []openrtb2.Imp) (openrtb_ext.BidType, error) {
+func getMediaTypeForImp(impID string, imps []openrtb2.Imp, bid visxBid) (openrtb_ext.BidType, error) {
 	for _, imp := range imps {
 		if imp.ID == impID {
+			var ext visxBidExt
+			if err := json.Unmarshal(bid.Ext, &ext); err == nil {
+				if ext.Prebid.Meta.MediaType == openrtb_ext.BidTypeBanner {
+					return openrtb_ext.BidTypeBanner, nil
+				}
+				if ext.Prebid.Meta.MediaType == openrtb_ext.BidTypeVideo {
+					return openrtb_ext.BidTypeVideo, nil
+				}
+			}
+
```

Comment:
From your PR description, it sounds like you want to favor setting the bid type to video if the imp has both banner and video objects. With this change I see your server will now be explicit about the bid type in its response, however, I see you're falling back to the implicit logic that checks if the `imp.Banner` is not nil in case the bid type was not specified, or is set but doesn't equal `BidTypeBanner` or `BidTypeVideo` for some reason. Do you want to prioritize setting the bid type to video in this situation by moving the block `if imp.Video != nil {` above the `if imp.Banner != nil {` block?

Reply:
Hi @bsardo, our developer is OOO, so I'll try to answer your question. Now VIS.X always returns 'banner' or 'video' in `Meta.MediaType` in bid responses to our Prebid Server adapter. Thus the code you mentioned above is going to return one of `BidTypeBanner` or `BidTypeVideo`. The code below the mentioned one is unlikely to be executed.

> Do you want to prioritize setting the bid type to video in this situation by moving the block if imp.Video != nil { above the if imp.Banner != nil { block?

The code is unlikely to be executed, I'd like to keep the current logic for now. When the developer comes back, we'll review this code and may make it clearer.